signer/xpi/xpi.go

```diff
@@ -142,20 +154,145 @@ func (s *PKCS7Signer) Config() signer.Configuration {
 	}
 }
 
-// SignFile takes an unsigned zipped XPI file and returned a signed XPI file
-func (s *PKCS7Signer) SignFile(input []byte, options interface{}) (signer.SignedFile, error) {
+// isCOSEAlgBad checks whether a COSE algorithm is recognized as a
+// valid IANA algorithm name, and supported and enabled in autograph
+func isCOSEAlgBad(algName string) (alg *cose.Algorithm, err error) {
+	alg, algErr := cose.GetAlgByName(algName)
+	if algErr != nil {
+		err = errors.Wrapf(algErr, "xpi: unrecognized algorithm COSE Signing algorithm")
+		return
+	}
+	if _, exists := supportedCOSEAlgNames[alg.Name]; exists == false {
```

Comment:
totally a nit, but given there are only 4 values we accept, wouldn't an if statement with string equality be more readable?

Reply:
Yeah, I was thinking it'd be easier to add new algorithms.